models/neural_nets/rnn_word2vec_with_question_classifier.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+import numpy as np
+from keras import Input, Model
+from keras.layers import Dropout, Dense, merge, concatenate
+
+from models.neural_nets.neural_net_classifier import NeuralNetClassifier
+from utils.utils import transform_sentence_batch_to_vector, lower_text, process_html, \
+    THREAD_FEATURES_COUNT, get_lstm
+
+
+class RnnWord2VecWithQuestionClassifier(NeuralNetClassifier):
+    def __init__(self, question_title_words_count, question_body_words_count, answer_body_words_count, lstm_embed_size,
+                 hidden_layer_size, bidirectional, dropout, mode):
+        super().__init__()
+        logging.info('RNN Word2Vec with question classifier')
+        logging.info(
+            'question_title_words_count = {}, question_body_words_count = {}, answer_body_words_count = {}, lstm_embed_size = {}, hidden_layer_size = {}, bidirectional = {}, dropout = {}, mode = {}'.format(
+                question_title_words_count, question_body_words_count, answer_body_words_count, lstm_embed_size,
+                hidden_layer_size, bidirectional, dropout, mode))
+
+        self.question_title_words_count = question_title_words_count
+        self.question_body_words_count = question_body_words_count
+        self.answer_body_words_count = answer_body_words_count
+
+        question_title_input = Input(shape=(self.question_title_words_count, self.num_features),
+                                     name='question_title_input')
+        question_body_input = Input(shape=(self.question_body_words_count, self.num_features),
+                                    name='question_body_input')
+        answer_body_input = Input(shape=(self.answer_body_words_count, self.num_features),
+                                  name='answer_body_input')
+        linguistic_features = Input(shape=(self.linguistic_features_calculator.LINGUISTIC_FEATURES_COUNT,),
+                                    name='linguistic_features')
+        thread_features = Input(shape=(THREAD_FEATURES_COUNT,), name='thread_features')
+
+        question_title_rnn_features = get_lstm(lstm_embed_size, bidirectional, dropout)(question_title_input)
```

Comment:
Network seems extremely wide

Reply:
Separate rnn's for each input is overkill
Try concatenating question + answer and decreasing dimensionality before recurrent layer